index.d.ts

```diff
@@ -130,14 +135,15 @@ export class VM {
  * to any VM (context); rather, it is bound before each run, just for that run.
  */
 export class VMScript {
-  constructor(code: string, path?: string, options?: {
-    lineOffset: number;
-    columnOffset: number;
-  });
-  /** Wraps the code */
-  wrap(prefix: string, postfix: string): VMScript;
-  /** Compiles the code. If called multiple times, the code is only compiled once. */
-  compile(): any;
+  code: string;
+  filename: string;
+  options: VMScriptOptions;
```

Comment:
Everything should be readonly. `code` can be written to for backwards compatiblity, but new code shouldn't do that.